s3tests_boto3/functional/test_s3.py

```diff
@@ -5102,6 +5102,107 @@ def test_bucket_recreate_not_overriding():
     objs_list = get_objects_list(bucket_name)
     eq(key_names, objs_list)
 
+
+@attr(resource='bucket')
+@attr(method='get')
+@attr(operation='list all objects in a bucket using list_objects_v2')
+@attr(assertion='returns all expected objects')
+
+def test_listobjects_v2():
+    client = get_client()
+    bucket_names = []
+    for i in xrange(5):
+        bucket_name = get_new_bucket_name()
+        bucket_names.append(bucket_name)
+
+    for name in bucket_names:
+        client.create_bucket(Bucket=name)
+
+    for i in bucket_names:
+    	for j in range(5):
+    		client.put_object(Bucket=i, Key=str(j))
+
+    response = client.list_buckets()
+    bucket_dicts = response['Buckets']
+    buckets_list = []
+
+    buckets_list = get_buckets_list()
+
+    for name in bucket_names:
+        if name in buckets_list:
+            response1 = client.list_objects_v2(Bucket=name)
```

Comment:
@albIN7 the response that this omits is already a dict. You can look for values in the dict like `KeyCount` and use `assert_raises()` to to check for the KeyCount value without raising your own ValueError. This same principle applies to the other tests you have as well.